lib/puppet_litmus/rake_tasks.rb

```diff
@@ -182,23 +185,43 @@
     raise "Source folder doesnt exist #{source_folder}" unless File.directory?(source_folder)
 
     module_tars = build_modules_in_folder(source_folder)
-    puts 'Building'
-    module_tars.each do |module_tar|
-      print "#{File.basename(module_tar)} "
-    end
     require 'bolt_spec/run'
     include BoltSpec::Run
-    puts "\nSending"
-    module_tars.each do |module_tar|
-      upload_file(module_tar.path, "/tmp/#{File.basename(module_tar)}", target_nodes, options: {}, config: nil, inventory: inventory_hash)
-      print "#{File.basename(module_tar)} "
-    end
-    puts "\nInstalling"
+    puts '\nInstalling'
     module_tars.each do |module_tar|
+      module_name = File.basename(module_tar)
+      local_path = "#{module_tar.path}"
+      remote_path = "/tmp/#{module_name}"
+      install_module_command = "puppet module install --force /tmp/#{module_name}"
+      puts "uploading and installing module #{module_name} to #{remote_path} on target(s) #{target_nodes}"
+      # upload module
+      upload_result = upload_file(
+        local_path,
+        remote_path,
+        target_nodes,
+        options: {},
+        config: nil,
+        inventory: inventory_hash.clone, # clone inventory due to bug in Bolt as mentioned above
```

Comment:
Clone is only a shallow copy.

Reply:
This should fix mangling of version key in inventory therefore shallow copy is sufficient for that.

Reply:
This was fixed in bolt v2.0.1 and we can remove the workaround locally once #254 is merged

Reply:
#254 will need to have the lower version of bolt updated to 2.0.1